### PR TITLE
remove usage of `BOOST_AUTO` and `PMACC_AUTO`

### PR DESCRIPTION
**Pull request by psychocoderHPC**
follow up of  #1743

remove `BOOST_AUTO` and `PMACC_AUTO` with native C++11 `auto` keyword

### CODE REVIEW

File: src/picongpu/include/fields/MaxwellSolver/DirSplitting/DirSplitting.hpp [hunk 1]
```diff
@@ -101,11 +101,11 @@ class DirSplitting : private ConditionCheck<fieldSolver::FieldSolver>
         FieldE& fieldE = dc.getData<FieldE > (FieldE::getName(), true);
         FieldB& fieldB = dc.getData<FieldB > (FieldB::getName(), true);
 
-        BOOST_AUTO(fieldE_coreBorder,
+        auto fieldE_coreBorder =
             fieldE.getGridBuffer().getDeviceBuffer().
                    cartBuffer().view(GuardDim().toRT(),
                                      -GuardDim().toRT()));
```
Comment (ax3l) [inline src/picongpu/include/fields/MaxwellSolver/DirSplitting/DirSplitting.hpp]: 2x: remove `)` to fix compile :)

File: src/libPMacc/test/memory/HostBufferIntern/setValue.hpp [hunk 2]
```diff
@@ -47,7 +47,7 @@ struct setValueTest
             const Data value = 255;
             hostBufferIntern.setValue(value);
 
-            PMACC_AUTO( ptr, hostBufferIntern.getPointer( ) );
```
Comment (ax3l) [inline src/libPMacc/test/memory/HostBufferIntern/setValue.hpp]: `PMACC_AUTO`: this part of the code seems not to be covered by the compile suite

**Comment by psychocoderHPC** [comment]
updated

File: src/libPMacc/test/memory/HostBufferIntern/setValue.hpp (hunk 2 above)
Comment (ax3l) [inline src/libPMacc/test/memory/HostBufferIntern/setValue.hpp]: ah, we need to build the tests (soon)

File: src/libPMacc/test/particles/IdProvider.hpp [hunk 3]
```diff
@@ -124,7 +124,7 @@ struct IdProviderTest
                 (idBuf.getDeviceBuffer().getDataBox(), numThreads, numIdsPerThread);
         idBuf.deviceToHost();
         BOOST_REQUIRE_EQUAL(numIds, ids.size());
-        PMACC_AUTO(hostBox, idBuf.getHostBuffer().getDataBox());
```
Comment (ax3l) [inline src/libPMacc/test/particles/IdProvider.hpp]: will add IDs to the particles in Bunch

**Pull request merged**